packages/genesys-spark-components/src/components/beta/gux-selector-cards/gux-selector-card/gux-selector-card.tsx

```diff
@@ -0,0 +1,143 @@
+import { Component, h, JSX, Listen, Element, Prop, State } from '@stencil/core';
+
+import { trackComponent } from '@utils/tracking/usage';
+import {
+  validateFormIds,
+  getSlottedInput
+} from '../../../stable/gux-form-field/gux-form-field.service';
+import { GuxSelectorCardVariant } from './gux-selector-card.types';
+import { calculateInputDisabledState } from '@utils/dom/calculate-input-disabled-state';
+import { onInputDisabledStateChange } from '@utils/dom/on-input-disabled-state-change';
+import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
+import { logWarn } from '@utils/error/log-error';
+import { setSlotAriaDescribedby } from '../../../stable/gux-form-field/gux-form-field.service';
+
+/**
+ * @slot icon - Required slot for icon
+ * @slot input - Required slot for input tag
+ * @slot label - Required slot for label tag
+ * @slot content - Optional slot for additional text content
+ * @slot badge - Optional slot for badge
+ */
+
+@Component({
+  styleUrl: 'gux-selector-card.scss',
+  tag: 'gux-selector-card-beta',
+  shadow: true
+})
+export class GuxSelectorCard {
+  private input: HTMLInputElement;
+  private disabledObserver: MutationObserver;
+
+  @Element()
+  private root: HTMLElement;
+
+  //TODO: COMUI-2784 Descriptive variant styling
+  @Prop()
+  variant: GuxSelectorCardVariant = 'simple';
+
+  @State()
+  private disabled: boolean;
+
+  @State()
+  private checked: boolean;
+
+  componentWillLoad(): void {
+    this.setInput();
+    trackComponent(this.root, { variant: this.variant });
+    this.validateLabelSlot(this.root.querySelector('[slot=label]'));
+  }
+
+  disconnectedCallback(): void {
+    if (this.disabledObserver) {
+      this.disabledObserver.disconnect();
+    }
+  }
+
+  @Listen('click')
+  onClick(e: MouseEvent): void {
+    e.stopPropagation();
+    this.setChecked();
+  }
+
+  private setInput(): void {
+    this.input = getSlottedInput(
+      this.root,
+      'input[type="radio"][slot="input"]'
+    );
+    this.checked = this.input.checked;
+    preventBrowserValidationStyling(this.input);
+
+    setSlotAriaDescribedby(this.root, this.input, 'content');
+
+    this.disabled = calculateInputDisabledState(this.input);
+
+    this.disabledObserver = onInputDisabledStateChange(
+      this.input,
+      (disabled: boolean) => {
+        this.disabled = disabled;
+      }
+    );
+
+    validateFormIds(this.root, this.input);
+  }
+
+  private validateLabelSlot(slottedElement: HTMLLabelElement): void {
+    if (slottedElement) {
+      const slottedTagName = slottedElement.tagName;
+      if (slottedTagName === 'LABEL') {
+        return;
+      }
+      logWarn(this.root, `You must slot an label in the label slot.`);
+    }
+  }
+
+  private setChecked(): void {
+    if (this.disabled) {
+      return;
```

Comment:
I think to avoid this disabled check you could style the host element like with `pointer-events:none` when the gux-disabled class is applied. I think something like this is done in other components.

Reply:
Updated to apply this approach.